sources/healthdog/src/main_test.rs

```diff
@@ -0,0 +1,208 @@
+use crate::error::{Error, Result};
+use crate::main_inner;
+use crate::service_check::{ServiceCheck, ServiceHealth};
+use httptest::responders::status_code;
+use httptest::{matchers::*, Expectation, Server};
+use std::fs::write;
+use std::path::PathBuf;
+use tempfile::TempDir;
+
+const OS_RELEASE: &str = r#"PRETTY_NAME=Bottlerocket,
+VARIANT_ID=myvariant
+VERSION_ID=1.2.3
+BUILD_ID=abcdef0
+"#;
+
+struct MockCheck {}
+
+impl ServiceCheck for MockCheck {
+    fn check(&self, service_name: &str) -> Result<ServiceHealth> {
+        if service_name.ends_with("failed") {
+            Ok(ServiceHealth {
+                is_healthy: false,
+                exit_code: Some(1),
+            })
+        } else {
+            Ok(ServiceHealth {
+                is_healthy: true,
+                exit_code: None,
+            })
+        }
+    }
+}
+
+// dynamically create a config file where we can set server port, list of services, and send_metrics
+fn create_config_file_contents(port: u16, services: &[&str], send_metrics: bool) -> String {
+    let mut svcs = String::new();
+    for (i, &service) in services.iter().enumerate() {
+        if i == 0 {
+            svcs.push('"')
+        }
+        svcs.push_str(service);
+        if i == services.len() - 1 {
+            svcs.push('"');
+        } else {
+            svcs.push_str("\",\"");
+        }
+    }
```

Comment:
Not quite, but this will work (will push soon):

```rust
    let svcs = services
        .iter()
        .map(|&s| format!("\"{}\"", s))
        .collect::<Vec<String>>()
        .join(", ");
```

Reply:
Implemented the more functional version in [2a5aef0](https://github.com/bottlerocket-os/bottlerocket/compare/f515d383c9e181c8a531a02414b320ff9231a81f..2a5aef04d7abcae5c91bf8cec84b8e95c7ccd306)